walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -34,23 +35,70 @@
     }
 
     // Seperate relevant TimeRanges from events, put into ArrayList and sort by ascending meeting start time
+    // Optional attendees are handled by running two sets of queries in parallel: a query if we are including optional attendees,
+    // and a query if we are only considering required attendees. If the optional attendee query is empty, then that means that 
+    // there are no valid times if we include optional attendees, and optional attendees are ignored.
     List<TimeRange> attendedMeetings = new ArrayList<>();
+    List<TimeRange> allAttendedMeetings = new ArrayList<>();
     for (Event event : events) {
       // First check if the event in question contains people from request, add those meetings to attendedMeetings
-      Set<String> attendees = new HashSet<>(request.getAttendees());
-      attendees.retainAll(event.getAttendees());
+      // Attendees is only required attendees, allAttendees includes optional attendees
+      Set<String> requiredAttendees = new HashSet<>(request.getAttendees());
+      Set<String> optionalAttendees = new HashSet<>(request.getOptionalAttendees());
+      Set<String> allAttendees = new HashSet<>(Sets.union(requiredAttendees, optionalAttendees));
+      
+      // I couldn't think of a way to factor this, I'm sorry Zu :'(
```

Comment:
This is looks fine with me. Please remove this comment.

Reply:
Oops done!